pkg/ingestor/parser/cyclonedx/parser_cyclonedx_test.go

```diff
@@ -492,7 +492,31 @@ func Test_cyclonedxParser_getComponentPackages(t *testing.T) {
 				Type: cdx.ComponentTypeFile,
 			}},
 		},
-		wantPurl: "pkg:guac/files/home/work/test/build/webserver",
+		wantPurl: []string{"pkg:guac/files/home/work/test/build/webserver"},
+	}, {
+		name: "nested components",
```

Comment:
Could you also add this to the unit test: `Test_cyclonedxParser` so that we can see that the `isDependency` predicates are defined properly per the dependencies defined in the SBOM:


```
"dependencies": [
    {
      "ref": "pkg:oci/mandrel-for-jdk-21-rhel8@sha256%3A41d92dafa5ccbf7f76fa81c5a0e7de83c51166f27bea9b98df018f644016bf04?arch=amd64&os=linux&tag=23.1-13.1724180416",
      "dependsOn": [
        "pkg:rpm/redhat/abattis-cantarell-fonts@0.0.25-6.el8?arch=noarch",
        "pkg:maven/compiler/compiler@23.1.4.0-1-redhat-00001?type=jar"
      ]
    },
    {
      "ref": "pkg:rpm/redhat/abattis-cantarell-fonts@0.0.25-6.el8?arch=noarch",
      "dependsOn": []
    },
    {
      "ref": "pkg:maven/compiler/compiler@23.1.4.0-1-redhat-00001?type=jar",
      "dependsOn": []
    }
  ]
```

Reply:
Good point, thanks :+1: 
I've added both `CycloneDX nested components` and `CycloneDX flat components` tests providing in input two SBOMs with the same information: the former with nested components and the latter with a flat array. The relevant part for these tests is that both expected the same `ociComponentsIsDependencyIngests` value for the `IsDependency` field basically ensuring consistency among the two approaches (nested vs flat).